Kentico.Kontent.Management/Models/LanguageVariants/Elements/BaseElement.cs

```diff
@@ -21,34 +25,140 @@ public abstract class BaseElement
     public abstract dynamic ToDynamic();
 
     /// <summary>
-    /// Get dynamic representation of the element reference.
+    /// Transforms the dynamic object to the element.
     /// </summary>
-    protected dynamic GetDynamicReference()
+    public static BaseElement FromDynamic(dynamic source, ElementMetadataType elementType) => elementType switch
     {
-        if (Element.Id != null)
+        ElementMetadataType.Text => FromDynamic(source, typeof(TextElement)),
+        ElementMetadataType.RichText => FromDynamic(source, typeof(RichTextElement)),
+        ElementMetadataType.Number => FromDynamic(source, typeof(NumberElement)),
+        ElementMetadataType.MultipleChoice => FromDynamic(source, typeof(MultipleChoiceElement)),
+        ElementMetadataType.DateTime => FromDynamic(source, typeof(DateTimeElement)),
+        ElementMetadataType.Asset => FromDynamic(source, typeof(AssetElement)),
+        ElementMetadataType.LinkedItems => FromDynamic(source, typeof(LinkedItemsElement)),
+        ElementMetadataType.Taxonomy => FromDynamic(source, typeof(TaxonomyElement)),
+        ElementMetadataType.UrlSlug => FromDynamic(source, typeof(UrlSlugElement)),
+        ElementMetadataType.Custom => FromDynamic(source, typeof(CustomElement)),
+        ElementMetadataType.Subpages => FromDynamic(source, typeof(SubpagesElement)),
+        _ => throw new InvalidOperationException("Unknown element type"),
+    };
+
+    /// <summary>
+    /// Transforms the dynamic object to the element.
+    /// </summary>
+    public static BaseElement FromDynamic(dynamic source, Type type)
+    {
+        ValidateSource(source);
```

Comment:
Even though there is validation in place, dynamic still can have an unexpected shape, or the value property might have an unexpected type. I suggest that we put the whole FromDynamic into try-catch block and throw an exception with a message like "Object could not be converted to the strongly-typed element. Please check if it has expected properties with expected type".